docs/usage/types.md

```diff
@@ -246,9 +246,20 @@ the first one that matches. In the above example the `id` of `user_03` was defin
 is defined under the attribute's `Union` annotation) but as the `uuid.UUID` can be marshalled into an `int` it
 chose to match against the `int` type and disregarded the other types.
 
+!!! warning
+    `typing.Union` also ignores order when [defined](https://docs.python.org/3/library/typing.html#typing.Union),
+    so `Union[int, float] == Union[float, int]` which can lead to unexpected behaviour
+    when combined with matching based on the `Union` type order.
```

Comment:
```suggestion
    when combined with matching based on the `Union` type order inside other type definitions, such as `List` and `Dict` types (because python treats these definitions as singletons).
 For example, `Dict[str, Union[int, float]] == Dict[str, Union[float, int]]` with the order based on the first time it was defined.
```
Just went back to #2835, the issue fully manifests as typing doesn't re-initialise Dict and List type definitions because of this so wanted to make it a bit clearer.